gerbil.evaluate/src/main/java/org/aksw/gerbil/evaluate/impl/ROCCurve.java

```diff
@@ -189,4 +200,35 @@ public String toString() {
         return builder.toString();
     }
 
+    public void addDiagonally(int stepsUp, int stepsRight) {
+        // If we start with a triangle we have to add (0,0)
+        Point last;
+        if (points.isEmpty()) {
+            last = new Point(0,0);
+            points.add(last);
+        } else {
+            last = points.get(points.size() - 1);
+        }
+        double x = 0.0;
+        rightStepsCount += stepsRight;
+        if (rightStepsCount >= falseStmts) {
+            // We want to end up at 1.0 so simply add it instead of trying to
+            // calculate it and end up with 0.999...
+            x = 1.0;
+        } else {
+            x = last.x + rightStepsCount * stepLengthRight;
+        }
+        double y = 0.0;
+        upStepsCount += stepsUp;
+        if (upStepsCount >= trueStmts) {
+            // We want to end up at 1.0 so simply add it instead of trying to
+            // calculate it and end up with 0.999...
+            y = 1.0;
+        } else {
+            y = last.y + upStepsCount * stepLengthUp;
+        }
+        points.add(new Point(x, y));
+        lastDir = DIRECTION.DIAGONALLY;
+    }
```

Comment:
In addition (may the predictedValue changes but it will still be diagonally) 
we could check if the last Direction was DIAGONALLY then we could remove the last step without hesitation and have few points as possible. 
```java
if(lastDir==DIRECTION.DIAGONALLY){
     points.remove(points.size()-1);
}
points.add...
```

Reply:
No, because you can have diagonal parts with different growth rate. If you remove the point between two different diagonals, you may remove a corner that would influence the AUC calculation :wink: